src/core/net/udp6.cpp

```diff
@@ -202,7 +207,7 @@ otError Udp::Connect(SocketHandle &aSocket, const SockAddr &aSockAddr)
 
     if (!aSocket.IsBound())
     {
-        SuccessOrExit(error = Bind(aSocket, aSocket.GetSockName()));
+        SuccessOrExit(error = Bind(aSocket, aSocket.GetSockName(), false));
```

Comment:
```suggestion
        SuccessOrExit(error = Bind(aSocket, aSocket.GetSockName(), true));
```
Should bind to Thread only by default?